components/forms/login/LoginForm.tsx

```diff
@@ -0,0 +1,44 @@
+"use client"
+
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+
+type LoginData = {
+  username: string,
+  password: string,
+}
+
+export default function LoginForm() {
+  const { replace } = useRouter();
+  const [error, setError] = useState("");
+  const { register, handleSubmit, formState: { isSubmitting } } = useForm<LoginData>({
+    defaultValues: {
+      username: "",
+      password: ""
+    },
+  });
+
+  const onSubmit = async (data: LoginData) => {
+    const response = await fetch("/api/login", {
+      method: "POST",
+      body: JSON.stringify(data)
+    })
+    if (!response.ok) {
+      const status = response.status;
+      setError((status >= 400 && status < 500) ? "Incorrect username or password" : "Something went wrong, try again later")
+      return;
+    }
+    setError("")
+    replace("/")
+  }
+
+  return <form className="flex flex-col gap-4 w-full" onSubmit={handleSubmit(onSubmit)}>
+    {error && !isSubmitting && <div className="alert alert-error text-sm"><span>{error}</span></div>}
```

Comment:
Should we have a common "Error" component across all forms?

Reply:
Yes, I'd agree.

It'll be fixed in a separate issue